### PR TITLE
Eof notification

### PR DESCRIPTION
**Pull request by maxtaco**
Small patch for EOF notifications for server.

### CODE REVIEW

**Comment by jzila** [comment]
Might want to merge in my PR first, then update this PR to match the updated API.

**Comment by maxtaco** [comment]
Of course.

**Comment by akalin-keybase** [comment]
This looks fine to me, but since this is the 3rd hook, we may want to refactor those hooks into a Delegate interface.

**Comment by jzila** [comment]
@akalin-keybase I've already done this. I'm in the process of trying to factor `DecodeNext` into a more clear paradigm, but given that it's exported, that's turning out to be more difficult.

**Comment by akalin-keybase** [comment]
Cool! Good to know.

File: errors.go [hunk 1]
```diff
@@ -45,9 +45,9 @@ func (p ProtocolNotFoundError) Error() string {
 	return "protocol not found: " + p.p
 }
 
-type EofError struct{}
+type EOFError struct{}
 
-func (e EofError) Error() string {
+func (e EOFError) Error() string {
```
Comment (maxtaco) [inline errors.go]: BTW, I also meant to get rid of this class, in favor of `io.EOF`

**Comment by maxtaco** [comment]
[Here](https://github.com/keybase/client/blob/issue_1050_3/go/kex2/provisionee.go#L87-L106)'s my use case for it --- when the `provisioner` interrupts the protocol halfway through, the `provisionee` should abort too.

**Comment by jzila** [comment]
We've already merged the equivalent functionality for this.

In fact, [here](https://github.com/keybase/client/blob/b789e2a7385638456a95c19d45303832a7f7fac1/go/kex2/provisionee.go#L102), we can now just call `srv.AddCloseListener(p.done)` instead of worrying about the hook.

**Pull request closed without merging**